src/util/interpolate.ts

```diff
@@ -10,41 +29,67 @@ export const interpolateFactory = (interpolationType: 'number'|'color'|'array'|'
     }
 };
 
-export function number(a: number, b: number, t: number) {
-    return (a * (1 - t)) + (b * t);
+function number(from: number, to: number, t: number): number {
+    return from + t * (to - from);
 }
 
-export function color(from: Color, to: Color, t: number) {
-    return new Color(
-        number(from.r, to.r, t),
-        number(from.g, to.g, t),
-        number(from.b, to.b, t),
-        number(from.a, to.a, t)
-    );
+const colorInterpolation = {
```

Comment:
Better to have this as a static method with switch case instead of an object that is called with a key, more readable, easier find and replace, and more performant.

Reply:
Hmm… I believe that the current implementation also has its advantages. From the fact that such an object exists, we have for free the type of supported interpolation color spaces:
https://github.com/maplibre/maplibre-style/blob/e679ba7e2bf31c5d7ed99110bae654c5d257beb2/src/util/interpolate.ts#L5
and an easy-to-maintain function to check if a space is supported:
https://github.com/maplibre/maplibre-style/blob/e679ba7e2bf31c5d7ed99110bae654c5d257beb2/src/util/interpolate.ts#L14-L16
An alternative could be:
```ts
const supportedInterpolationColorSpaces = ['rgb', 'hcl', 'lab'] as const; // an extra array
export type InterpolationColorSpace = typeof supportedInterpolationColorSpaces[number];
export function isSupportedInterpolationColorSpace(colorSpace: string): colorSpace is InterpolationColorSpace {
    return supportedInterpolationColorSpaces.includes(colorSpace as InterpolationColorSpace);
}
```
or another, high-maintenance version but the most readable:
```ts
export type InterpolationColorSpace = 'rgb' | 'hcl' | 'lab';
export function isSupportedInterpolationColorSpace(colorSpace: string): colorSpace is InterpolationColorSpace {
    return colorSpace === 'rgb' || colorSpace === 'hcl' || colorSpace === 'lab';
}
```
I do not know, in my opinion it is fine as it is now, but if you think otherwise I can change it to swich case.

Reply:
Yes, please change to the last option, I find it more readable.

Reply:
Done